### PR TITLE
Add display_name in Config.

### PR DESCRIPTION
**Pull request by Krout0n**
SSIA.  This PR includes nit fixes too.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -35,9 +35,10 @@ async fn initialize(client: &Client, conf: &Config) -> std::io::Result<String> {
         let hostname = hostname.unwrap().to_str().unwrap().to_owned();
         let meta = mackerel_agent::host_meta::collect_as_json();
         let param = mackerel_client::create_host_param!({
-            name -> format!("{}.rs", hostname) // TODO .rs を付けてるのはは暫定的
+            name -> format!("{}.rs", hostname) // TODO: Remove .rs
```
Comment (ne-sachirou) [inline src/main.rs:38]: 👍

File: src/config.rs [hunk 2]
```diff
@@ -1,24 +1,30 @@
...
     pub apibase: String,
     pub apikey: String,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub roles: Option<Vec<String>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub display_name: Option<String>,
 }
 
-impl Config {
-    pub fn new() -> Self {
```
Comment (ne-sachirou) [inline src/config.rs:14]: [note] This is deprecated by `std::default::Default::default()` in the added lines in this pull request.

File: src/config.rs [hunk 3]
```diff
@@ -1,24 +1,30 @@
 use serde::Deserialize;
 use std::{fs, path::Path};
 
-#[derive(Default, Debug, Deserialize, PartialEq)]
+#[derive(Debug, Deserialize, PartialEq)]
+#[serde(default)]
 pub struct Config {
     #[serde(default = "Config::default_apibase")]
```
Comment (ne-sachirou) [inline src/config.rs]: Is this needed after adding the `#[serde(default)]`?

Comment (Krout0n) [reply src/config.rs]: Yes, the annotation is the mark for using the type's implementation of `std::default::Default`. So, I fix the code as only using `#[serde(default)]` in toplevel. https://github.com/mackerelio-labs/mackerel-agent-rs/pull/20/commits/dafa2963ffc8dc70cbd851328ce88e786dfbb068

**Review by ne-sachirou — APPROVED** [review]
🙆‍♀️ LGTM

**Comment by Krout0n** [comment]
Thank you:pray:

**Pull request merged**